src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,233 @@
+        if (line.contains("/by")) {
+            // Adding a Deadline
+            try {
+                addDeadline(line, tasks, taskCount);
+            } catch (IllegalDeadlineException e) {
+                deadlineErrorMessage();
+                throw new IllegalTaskException();
+            }
+        } else if (line.contains("/from") || line.contains("/to")) {
+            // Adding an Event
+            try {
+                addEvent(line, tasks, taskCount);
+            } catch (IllegalEventException e) {
+                eventErrorMessage();
+                throw new IllegalTaskException();
+            } catch (IndexOutOfBoundsException e) {
+                eventErrorMessage();
+                throw new IllegalTaskException();
+            }
+        } else if (!line.isBlank()){
+            // Adding a _Todo_
+            try {
+                addTodo(line, tasks, taskCount);
+            } catch (IllegalTodoException e) {
+                todoErrorMessage();
+                throw new IllegalTaskException();
+            }
+        } else {
+            throw new IllegalTaskException();
+        }
+    }
+
+    private static void todoErrorMessage() {
+        borderLine();
+        System.out.println("\t Error. Please enter a valid description.");
+        borderLine();
+    }
+
+    private static void eventErrorMessage() {
+        borderLine();
+        System.out.println("\t Error. Please enter a valid description, start time and end time");
+        borderLine();
+    }
+
+    private static void deadlineErrorMessage() {
+        borderLine();
+        System.out.println("\t Error. Please enter a valid description and deadline.");
+        borderLine();
+    }
+
+    private static void addTodo(String line, Task[] tasks, int taskCount) throws IllegalTodoException {
+        if (line.isBlank()) {
+            throw new IllegalTodoException();
+        } else {
+            tasks[taskCount] = new Todo(line);
+            addedTaskMessage(tasks[taskCount], taskCount);
+        }
+    }
+
+    private static void addEvent(String line, Task[] tasks, int taskCount) throws IllegalEventException {
+        String description = line.substring(0, line.indexOf("/from")).trim();
+        String start = line.substring(line.indexOf("/from") + 5, line.indexOf("/to")).trim();
+        String end = line.substring(line.indexOf("/to") + 3).trim();
+        if (description.isBlank() || start.isBlank() || end.isBlank()) {
+            throw new IllegalEventException();
+        } else {
+            tasks[taskCount] = new Event(description, start, end);
+            addedTaskMessage(tasks[taskCount], taskCount);
+        }
+    }
+
+    private static void addDeadline(String line, Task[] tasks, int taskCount) throws IllegalDeadlineException {
+        String description = line.substring(0, line.indexOf("/by")).trim();
+        String deadline = line.substring(line.indexOf("/by") + 3).trim();
```

Comment:
The trimming lines are similar to the ones in line 131 & 133, can consider abstracting it out and passing a variable (`/by` or `/from`) into the method